### ISSUE TITLE
rgbgfx fails when built with _GLIBCXX_ASSERTIONS

### ISSUE THREAD

**Issue by einstein95** [gbdev/rgbds#1102]
Noticed that the Arch Linux pkg was failing when compiling pokered with this error when doing `Png::isSuitableForGrayscale`
```
/usr/include/c++/12.2.0/optional:484: constexpr const _Tp& std::_Optional_base_impl<_Tp, _Dp>::_M_get() const [with _Tp = Rgba; _Dp = std::_Optional_base<Rgba, true, true>]: Assertion 'this->_M_is_engaged()' failed.
[1]    97322 IOT instruction (core dumped)  rgbgfx -o gfx/tilesets/flower/flower1.2bpp gfx/tilesets/flower/flower1.png
```
Traced it down to it being compiled with `-D_GLIBCXX_ASSERTIONS`, removing it and rgbgfx runs fine.

**Comment by einstein95** [comment]
git bisect traces the bug to 3fa1854332735f4817bd43eee748b44cdc811c06

**Comment by ISSOtm** [comment]
Yes, this has since been fixed in `master` by 12ba057b4f74a6ef4989d0ceb4075ef74bcc6ea2, and `_GLIBCXX_ASSERTIONS` (indirectly) added to CI in the subsequent 28737d5778f05bd554e595360c1f94d21328d15d.

I'm not sure whether we should publish v0.6.1 soon (this is fairly major breakage considering the project in question, *but* limited to a few distributions); in the meantime, I can suggest using a `master` build, potentially from the `rgbds-git` AUR package.

Regardless, thank you for taking the time to report the bug *and* bisect it!
